code/math/vecmat.cpp

```diff
@@ -2509,29 +2509,32 @@ void vm_matrix_sub2(matrix* dest, const matrix* src)
 */
 bool vm_inverse_matrix(matrix* dest, const matrix* m)
 {
-	matrix inv;	// create a temp matrix so we can avoid getting a determinant that is 0
+	// Use doubles here because this is used for ship inv_mois and we could be dealing with extremely small numbers
+	double inv[3][3];	// create a temp matrix so we can avoid getting a determinant that is 0
```

Comment:
You could do this:
```
union {
    double a2d[3][3];
    double a1d[9];
} inv;
```
That would allow you to keep the a2d and a1d references from before.

The main reason to do this is to keep the single loop instead of the nested loop.  This would be faster, but I don't think this is speed-critical code (unless you're inverting matrixes every frame somewhere) so it's optional.

Reply:
It might be slightly faster, but I think it's easier to understand if the code itself mimics the actual structure of the matrix.